src/gpu/symmetry.jl

```diff
@@ -0,0 +1,47 @@
+using GPUArraysCore
+
+function accumulate_over_symmetries!(ρaccu::AbstractGPUArray, ρin::AbstractGPUArray,
+                                     basis::PlaneWaveBasis{T}, symmetries) where {T}
+    if all(isone, symmetries)
+        ρaccu .+= ρin
+        return ρaccu
+    end
+
+    Gs = vec(G_vectors(basis))
+    ρtmp = similar(ρaccu)
+    fft_size = basis.fft_size
+
+    for symop in symmetries
+        if isone(symop)
+            ρaccu .+= ρin
+            continue
+        end
+
+        invS = Mat3{Int}(inv(symop.S))
+        map!(ρtmp, Gs) do G
+            idx = index_G_vectors(fft_size, invS * G)
+            isnothing(idx) ? zero(complex(T)) : cis2pi(-T(dot(G, symop.τ))) * ρin[idx]
+        end
```

Comment:
Apart from these 5 lines the function is basically identical to the CPU version. I'm wondering if it's not better to just duplicate these 5 lines into a GPU version as opposed to the entire `accumulate_over_symmetries!` function.

Reply:
It might not be so relevant anymore with the latest commit. The functions look more different now.